src/app.controller.ts

```diff
@@ -46,25 +56,69 @@ export class AppController {
     } = this.configService;
     if (isDevEnv) {
       const host = req.header('host');
+      const decodedIdToken = await this.authService.checkSession(req);
+
+      if (!decodedIdToken) {
+        return {
+          loginSuccessUrl: `http://${host}/login-success`,
+          postLoginSuccessUrl: `http://${host}/profile`,
+          apiKey: FirebaseAPIKey,
+          appId: FirebaseAppId,
+          authDomain: FirebaseAuthDomain,
+          databaseURL: FirebaseDbUrl,
+          messagingSenderId: FirebaseMessagingSenderId,
+          projectId: FirebaseProjectId,
+          storageBucket: FirebaseStorageBucket,
+        };
+      } else {
+        return res.redirect(301, `http://${host}/profile`);
+      }
+    } else {
+      throw new MethodNotAllowedException();
+    }
+  }
 
+  @Get('/profile')
+  @UseGuards(AuthGuard)
+  @Render('profile')
+  async getProfilePage(
+    @Req() req: Request,
+    @Res() res: Response,
+  ): Promise<GetProfilePageResponse | void> {
+    const decodedIdToken = await this.authService.checkSession(req);
+    const host = req.header('host');
+
+    if (decodedIdToken) {
       return {
-        loginSuccessUrl: `http://${host}/login-success`,
-        apiKey: FirebaseAPIKey,
-        appId: FirebaseAppId,
-        authDomain: FirebaseAuthDomain,
-        databaseURL: FirebaseDbUrl,
-        messagingSenderId: FirebaseMessagingSenderId,
-        projectId: FirebaseProjectId,
-        storageBucket: FirebaseStorageBucket,
+        email: decodedIdToken.email,
+        emailVerified: decodedIdToken.email_verified,
+        name: decodedIdToken.name,
+        picture: decodedIdToken.picture,
       };
     } else {
-      throw new MethodNotAllowedException();
+      return res.redirect(301, `http://${host}/login`);
     }
   }
 
   @Post('/login-success')
-  async handleNotify(@Res() res: Response, @Req() req: Request) {
-    console.error(req.body);
-    res.status(HttpStatus.OK).send();
+  async handleNotify(@Res() res: Response, @Req() req: Request): Promise<void> {
+    const { idToken, csrfToken, uid } = req.body;
+
+    if (csrfToken !== req.cookies['ps-csrfToken']) {
+      throw new UnauthorizedException('UNAUTHORIZED REQUEST!');
+    }
+
+    try {
+      const {
+        sessionCookie,
+        sessionOptions,
+      } = await this.authService.createSessionCookie(idToken);
+
+      res.cookie('ps-session', sessionCookie, sessionOptions);
+      res.cookie('ps-uid', uid);
+      res.end(JSON.stringify({ status: 'success' }));
+    } catch (_e) {
```

Comment:
why use underscore?

Reply:
good point. I left it since previously I just wrap code with try-catch. I will remove _. General speaking, _ can be used when You have more than one required parameter in the function, but actually want to work with for example second one (_first, second). Without _ eslint throws error that this param  is unused. With _ it omit this param.